### ISSUE TITLE
Use correct CA Root Certs in mongodb-enterprise-operator deployment

### ISSUE THREAD

**Issue by ron1** [mongodb/mongodb-enterprise-kubernetes#8]
CA Root Certs in the mongodb-enterprise-operator container should be injected in order for them to be accurate. A hostpath volume is a common mechanism for CA root injection within the Kubernetes community.

Since it appears the operator is written using the go language, the list of where go looks for root certs is here: https://golang.org/src/crypto/x509/root_linux.go.

In order to satisfy the go language requirements for find root certs, I suggest adding volumeMounts to the mongodb-enterprise-operator container as follows:

```yaml
    spec:
      containers:
      - name: mongodb-enterprise-operator
...
        volumeMounts:
        - mountPath: /etc/ssl
          name: ssl-folder
        - mountPath: /etc/pki
          name: pki-folder
...
      volumes:
      - name: ssl-folder
        hostPath:
          path: /etc/ssl
      - name: pki-folder
        hostPath:
          path: /etc/pki
```

Without this CA Root injection, the operator may not work in some environments due to ssl verification errors.

**Comment by antonlisovenko** [comment]
Thanks for suggestion, we planned to work on security aspects in the next quarters.

We haven't looked into this issue yet, do you suggest us to provide this configuration as the default one in https://github.com/mongodb/mongodb-enterprise-kubernetes/blob/master/mongodb-enterprise.yaml and in https://github.com/mongodb/mongodb-enterprise-kubernetes/blob/master/helm_chart/templates/operator.yaml? Should we stick to `hostPath` approach or may be we should leave this for the customers (may be they would prefer to have pool of static volumes with certificates placed on them and bind to them instead of having to put certificates to all hosts to survive rescheduling)?

**Comment by ron1** [comment]
My proposed hostpath-based solution is a simple, easy-to-implement improvement to the existing deployment. However, it only works if the container Linux distribution and the node Linux distribution are compatible since the SSL/TLS store location is not standardized across operating systems or even Linux distros (see https://gist.github.com/CMCDragonkai/f5f76b8eb13e7579aba3). This may be good enough until you can "work on security aspects in the next quarters". 

Your idea of using bind mounts with a pool of static volumes w/certs could be achieved by adding attribute "sslTrustedMMSServerCertificate" to the Credentials Secret. Here, the attribute value is the cert itself rather than a path. The Deployment would then mount this Secret attribute value (via a subPath) to the appropriate SSL/TLS store location for ca-bundle.crt. Of course, this would require a deployer to manually update "mongodb-enterprise.yaml" with the name given to the Credentials Secret.

Neither of these solutions should require changes to the operator Go source code. Your feedback is much appreciated.

**Comment by ron1** [comment]
My proposed hostpath-based solution worked ok with the ubuntu-based image for the mongodb-enterprise-operator container after the mongodb-enterprise-operator service account was assigned the hostpath privilege. Unfortunately, the mongodb-enterprise-database pods launched by the Operator-generated Statefulset require a fix as well. Since the Statefulset is Operator-generated, it is unclear how this can be fixed without either Go code changes or docker image extensions to add a CA root.

**Comment by antonlisovenko** [comment]
Thanks Ron for this suggestion.
We are currently starting work on SSLs and will consider your idea.
I've created an internal ticket to track this one

**Comment by jamesbroadhead** [comment]
I think that this has been resolved with our 0.11 release with TLS support. 
Please reopen with some more details if not
